commands/build.go

```diff
@@ -434,7 +456,263 @@ func buildCmd(dockerCli command.Cli, rootOpts *rootOptions) *cobra.Command {
 			options.contextPath = args[0]
 			options.builder = rootOpts.builder
 			cmd.Flags().VisitAll(checkWarnedFlags)
-			return runBuild(dockerCli, options)
+			if !isExperimental() || !options.detach {
```

Comment:
I'm not sure I understand this condition :smile:

If `--detach` is set, then `BUILDX_EXPERIMENTAL` must be enabled, since `--detach` is an experimental option. I assume it's explicitly to run the build on the server, but I think we should always default to that in experimental mode - so I'm not sure if we need to keep the `detach` option?

Reply:
Currently `--detach` is supported only on linux. So could we preserve `--detach=false` option for non-linux platform?